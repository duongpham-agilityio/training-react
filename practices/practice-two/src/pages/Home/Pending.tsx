import { memo } from 'react';

// Components
import { CardSkeleton } from 'components';

// Styles
import homeStyles from 'pages/Home/index.module.css';

const Pending = () => (
  <div className={homeStyles.content}>
    <div className={homeStyles.grid}>
      {Array.from({ length: 6 }).map((_, index) => (
        <CardSkeleton key={index} />
      ))}
    </div>
  </div>
);

export default memo(Pending);
