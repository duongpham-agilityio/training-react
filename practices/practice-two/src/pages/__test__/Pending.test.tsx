// Mocks
import 'components/__test__/mocks/image.test';

import { render } from '@testing-library/react';

// Components
import Pending from 'pages/Home/Pending';

describe('Pending', () => {
  it('Match to snapshot', () => {
    const { container } = render(<Pending />);

    expect(container).toMatchSnapshot();
  });
});
