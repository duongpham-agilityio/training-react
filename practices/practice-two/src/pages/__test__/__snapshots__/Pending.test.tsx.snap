// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Match to snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
      <div
        class="undefined undefined"
        data-testid="card-skeleton"
      >
        <div>
          <h2
            class="undefined undefined undefined undefined"
          />
          <p
            class="undefined undefined"
          />
          <div>
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
            <button
              class="undefined undefined
  undefined undefined
  undefined "
              type="button"
            />
          </div>
        </div>
        <div>
          <p
            class="undefined undefined"
          />
          <div
            class="undefined undefined undefined"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
